### PR TITLE
[Merton] character limit on extra question fields

### PR DESCRIPTION
**Pull request by ludovic-tc**
fixes mysociety/societyworks#2823

Please check the following:

- [x] Whether this PR should include changes to any documentation, or the FAQ;
- [x] All cobrand-specific commits start their commit message with the cobrand in square brackets;
- [x] Is new functionality tested? CodeCov will warn you about the diff coverage, but won’t complain about e.g. new files;
- [x] Will cobrand-specific changes require additional work to ensure consistent behaviour on www.fixmystreet.com? 
- [ ] Are the changes tested for accessibility?
[skip changelog]

### CODE REVIEW

**Comment by ludovic-tc** [comment]
- [ ] fix for Merton cobrand
  - [ ] fix for fly-tipping evidence
  - [ ] fix for all additional question fields
- [ ] fix for fixmystreet.com

**Review by davea — CHANGES_REQUESTED** [review]
Works as it should, but please add a test. I don't think we've done custom validation on extra fields before so it'd be good to cover it.

File: .cypress/cypress/integration/merton.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    cy.contains('Merton Council');
+});
+it('does not allow >255 chars in additional question fields', function() {
+    cy.server();
+    cy.visit('http://merton.localhost:3001/report/new?longitude=-0.199218&latitude=51.405454');
+    cy.get('input[value="Flytipping"]').click();
+    cy.get('button').first().click();
+    var longstring = '';
+    while (longstring.length < 260) { longstring += 'a'; }
+    cy.get('textarea#form_evidence').type(longstring);
```
Comment (davea) [inline .cypress/cypress/integration/merton.js:12]: This could just be `cy.get('textarea#form_evidence').type("a".repeat(260));` :)

**Review by davea — CHANGES_REQUESTED** [review]
Nearly there, just a couple of small things to fix to get Cypress green.

File: .cypress/cypress/integration/merton.js (hunk 1 above)
Comment (ludovic-tc) [reply .cypress/cypress/integration/merton.js]: Hi Dave, thanks. I've incorporated this while also reusing this code for validation. But makes fewer LOCs and less repetition than before I saw this.

I also spotted that I'd added some redundant code. (The extra question that I got stuck on above, I'd actually already added separately in a way that worked! This is the trouble with putting stuff to one side for weeks.)

File: .cypress/cypress/integration/merton.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    var longstring260 = 'a'.repeat(260);
+    cy.get('input#form_evidence').as('evidence');
+    cy.get('@evidence').type('a'.repeat(260));
+    cy.get('@evidence').invoke('val').then(function(val){
+        expect(val.length).to.be.at.most(255);
+    });
+    cy.get('@evidence').invoke('val', longstring260);
+    cy.get('@evidence').invoke('val').then(function(val){
+        expect(val.length).to.equal(260);
+    });
```
Comment (davea) [inline .cypress/cypress/integration/merton.js:19]: I think these lines would make more logical sense in the next test (starting on L21) as they're setting up what's being tested there.

**Review by davea — CHANGES_REQUESTED** [review]
One very tiny tweak is all!

File: .cypress/cypress/integration/merton.js (hunk 2 above)
Comment (ludovic-tc) [reply .cypress/cypress/integration/merton.js:21]: ✅

File: .cypress/cypress/integration/merton.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    cy.get('button').first().click();
+    var longstring260 = 'a'.repeat(260);
+    cy.get('input#form_evidence').as('evidence');
+    cy.get('@evidence').type('a'.repeat(260));
+    cy.get('@evidence').invoke('val').then(function(val){
+        expect(val.length).to.be.at.most(255);
+    });
+});
+it('does not allow user to continue if additional question field value is >255', function() {
+    cy.get('@evidence').invoke('val', longstring260);
```
Comment (davea) [inline .cypress/cypress/integration/merton.js:18]: This is failing because `longstring260` isn't in scope here.

**Review by davea — CHANGES_REQUESTED** [review]
Don't forget to run the tests locally before submitting for review!

File: .cypress/cypress/integration/merton.js (hunk 3 above)
Comment (ludovic-tc) [reply .cypress/cypress/integration/merton.js:18]: Thanks Dave, sorry, I usually do do this, have added another reminder! 🤦🏼‍♂️

File: .cypress/cypress/integration/merton.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+it('loads the Merton FMS Pro front page', function() {
+    cy.visit('http://merton.localhost:3001/');
+    cy.contains('Merton Council');
+});
+it('does not allow typing >255 chars in additional question fields', function() {
+    //cy.server();
```
Comment (davea) [inline .cypress/cypress/integration/merton.js:6]: If this line isn't required can it be removed instead of left commented out?

**Review by davea — CHANGES_REQUESTED** [review]

File: .cypress/cypress/integration/merton.js (hunk 4 above)
Comment (ludovic-tc) [reply .cypress/cypress/integration/merton.js:6]: 🤦🏼

**Review by davea — APPROVED** [review]

**Review by davea — APPROVED** [review]

**Pull request merged**